src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioRazorParser.cs

```diff
@@ -0,0 +1,372 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.VisualStudio.Language.Intellisense;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Operations;
+using ITextBuffer = Microsoft.VisualStudio.Text.ITextBuffer;
+using Timer = System.Threading.Timer;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    internal class DefaultVisualStudioRazorParser : VisualStudioRazorParser, IDisposable
+    {
+        // Internal for testing.
+        internal TimeSpan IdleDelay = TimeSpan.FromSeconds(3);
+        internal Timer _idleTimer;
+
+        private readonly object IdleLock = new object();
+        private readonly ICompletionBroker _completionBroker;
+        private readonly IEnumerable<ICanHasContextChangedListener> _parserContextChangedListeners;
+        private readonly VisualStudioDocumentTracker _documentTracker;
+        private readonly ForegroundDispatcher _dispatcher;
+        private readonly RazorTemplateEngineFactoryService _templateEngineFactory;
+        private readonly string _projectDirectory;
```

Comment:
Is this part of the tracker? should it be?

Reply:
It kind of is via its `Project` Roslyn property but it explodes if accessed prior to the Roslyn project system being created. We could make an exception for this directory and make it always available. on the tracker.